File: conference/registration-bc/web/src/main/java/com/microsoft/conference/registration/commandhandlers/OrderCommandHandler.cs
using System.Linq;
using System.Threading.Tasks;
using ECommon.Components;
using ENode.Commanding;
using Registration.Commands.Orders;
using Registration.Orders;

namespace Registration.CommandHandlers
{
    [Component]
    public class OrderCommandHandler :
        ICommandHandler<PlaceOrder>,
        ICommandHandler<AssignRegistrantDetails>,
        ICommandHandler<ConfirmReservation>,
        ICommandHandler<ConfirmPayment>,
        ICommandHandler<MarkAsSuccess>,
        ICommandHandler<CloseOrder>
    {
        private readonly IPricingService _pricingService;

        public OrderCommandHandler(IPricingService pricingService)
        {
            _pricingService = pricingService;
        }

        public Task HandleAsync(ICommandContext context, PlaceOrder command)
        {
            return context.AddAsync(new Order(
                command.AggregateRootId,
                command.ConferenceId,
                command.Seats.Select(x => new SeatQuantity(new SeatType(x.SeatType, x.SeatName, x.UnitPrice), x.Quantity)),
                _pricingService));
        }
        public async Task HandleAsync(ICommandContext context, AssignRegistrantDetails command)
        {
            var order = await context.GetAsync<Order>(command.AggregateRootId);
            order.AssignRegistrant(command.FirstName, command.LastName, command.Email);
        }
        public async Task HandleAsync(ICommandContext context, ConfirmReservation command)
        {
            var order = await context.GetAsync<Order>(command.AggregateRootId);
            order.ConfirmReservation(command.IsReservationSuccess);
        }
        public async Task HandleAsync(ICommandContext context, ConfirmPayment command)
        {
            var order = await context.GetAsync<Order>(command.AggregateRootId);
            order.ConfirmPayment(command.IsPaymentSuccess);
        }
        public async Task HandleAsync(ICommandContext context, MarkAsSuccess command)
        {
            var order = await context.GetAsync<Order>(command.AggregateRootId);
            order.MarkAsSuccess();
        }
        public async Task HandleAsync(ICommandContext context, CloseOrder command)
        {
            var order = await context.GetAsync<Order>(command.AggregateRootId);
            order.Close();
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/commandhandlers/OrderCommandHandler.java
package com.microsoft.conference.registration.commandhandlers;

import com.microsoft.conference.common.registration.commands.orders.AssignRegistrantDetails;
import com.microsoft.conference.common.registration.commands.orders.CloseOrder;
import com.microsoft.conference.common.registration.commands.orders.ConfirmPayment;
import com.microsoft.conference.common.registration.commands.orders.ConfirmReservation;
import com.microsoft.conference.common.registration.commands.orders.MarkAsSuccess;
import com.microsoft.conference.common.registration.commands.orders.PlaceOrder;
import com.microsoft.conference.registration.domain.SeatQuantity;
import com.microsoft.conference.registration.domain.SeatType;
import com.microsoft.conference.registration.domain.order.IPricingService;
import com.microsoft.conference.registration.domain.order.models.Order;
import org.enodeframework.annotation.Command;
import org.enodeframework.annotation.Subscribe;
import org.enodeframework.commanding.ICommandContext;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.ArrayList;
import java.util.List;

import static org.enodeframework.common.io.Task.await;

@Command
public class OrderCommandHandler {

    @Autowired
    private IPricingService pricingService;

    @Subscribe
    public void handleAsync(ICommandContext context, PlaceOrder command) {
        List<SeatQuantity> seats = new ArrayList<>();
        command.seatInfos.forEach(x -> seats.add(new SeatQuantity(new SeatType(x.seatType, x.seatName, x.unitPrice), x.quantity)));
        context.addAsync(new Order(
                command.aggregateRootId,
                command.conferenceId,
                seats,
                pricingService));
    }

    @Subscribe
    public void handleAsync(ICommandContext context, AssignRegistrantDetails command) {
        Order order = await(context.getAsync(command.aggregateRootId, Order.class));
        order.assignRegistrant(command.firstName, command.lastName, command.email);
    }

    @Subscribe
    public void handleAsync(ICommandContext context, ConfirmReservation command) {
        Order order = await(context.getAsync(command.aggregateRootId, Order.class));
        order.confirmReservation(command.isReservationSuccess);
    }

    @Subscribe
    public void handleAsync(ICommandContext context, ConfirmPayment command) {
        Order order = await(context.getAsync(command.aggregateRootId, Order.class));
        order.confirmPayment(command.isPaymentSuccess);
    }

    @Subscribe
    public void handleAsync(ICommandContext context, MarkAsSuccess command) {
        Order order = await(context.getAsync(command.aggregateRootId, Order.class));
        order.markAsSuccess();
    }

    @Subscribe
    public void handleAsync(ICommandContext context, CloseOrder command) {
        Order order = await(context.getAsync(command.aggregateRootId, Order.class));
        order.close();
    }
}
